project_task_material/tests/test_create_material_lines.py

```diff
@@ -1,35 +1,89 @@
+
+        """
+        try:
+            # Material with `quantity = 0.0`
+            self.task.sudo(self.project_manager.id).write({"material_ids": [(
+                0, 0, {"product_id": self.product.id, "quantity": 0.0})]})
+            # Material with `negative quantity`
+            self.task.sudo(self.project_manager.id).write({"material_ids": [(
+                0, 0, {"product_id": self.product.id, "quantity": -10.0})]})
+            # Material with `positive quantity`
+            self.task.sudo(self.project_manager.id).write({"material_ids": [(
+                0, 0, {"product_id": self.product.id, "quantity": 3.0})]})
+        except ValidationError as err:
+            self.assertEqual(
+                str(err.args[0]),
+                "Quantity of material consumed must be greater than 0.")
+
+    def test_manager_add_task_material_right(self):
+        """
+        TEST CASE 2
+        The project manager is adding some materials in the task
+        with different right values
+
+        """
+        # Material with `quantity = 0.0`
+        self.task.sudo(self.project_manager.id).write({"material_ids": [(
+            0, 0, {"product_id": self.product.id, "quantity": 4.0})]})
+        # Material with `negative quantity`
+        self.task.sudo(self.project_manager.id).write({"material_ids": [(
+            0, 0, {"product_id": self.product.id, "quantity": 5.0})]})
+        # Material with `positive quantity`
+        self.task.sudo(self.project_manager.id).write({"material_ids": [(
+            0, 0, {"product_id": self.product.id, "quantity": 3.0})]})
+        self.assertEqual(len(self.task.material_ids.ids), 3)
+
+    def test_user_add_task_material_wrong(self):
+        """
+        TEST CASE 3
+        The project user is adding some materials in the task
+        with different wrong values
+
+        """
+        try:
+            # Material with `quantity = 0.0`
+            self.task.sudo(self.project_user.id).write({"material_ids": [(
+                0, 0, {"product_id": self.product.id, "quantity": 0.0})]})
+            # Material with `negative quantity`
+            self.task.sudo(self.project_user.id).write({"material_ids": [(
+                0, 0, {"product_id": self.product.id, "quantity": -10.0})]})
+            # Material with `positive quantity`
+            self.task.sudo(self.project_user.id).write({"material_ids": [(
+                0, 0, {"product_id": self.product.id, "quantity": 3.0})]})
+        except ValidationError as err:
+            self.assertEqual(
+                str(err.args[0]),
+                "Quantity of material consumed must be greater than 0.")
+
+    def test_user_add_task_material_right(self):
+        """
+        TEST CASE 4
+        The project user is adding some materials in the task
+        with different right values
+
+        """
+        # Material with `quantity = 0.0`
+        self.task.sudo(self.project_user.id).write({"material_ids": [(
```

Comment:
Save a variable with `task = self.task.sudo(self.project_user.id)` to avoid the same call too much times.